graph_generation/test_attribute_generator.py

```diff
@@ -0,0 +1,76 @@
+"""
+Module to test graph attribute generation.
+
+Usage:
+    python3 graph_generation/test_attribute_generator.py
+"""
+
+import unittest
+from attribute_generator import AttributeGenerator
+
+from config_params.collection_params import CollectionParams
+from config_params.data_integrity_params import DataIntegrityParams
+from config_params.dataset_params import DatasetParams
+from config_params.dataset_to_system_params import DatasetToSystemParams
+from config_params.processing_params import ProcessingParams
+from config_params.system_params import SystemParams
+from config_params.connection_params import ConnectionParams
+
+
+class TestAttributeGenerator(unittest.TestCase):
+    @staticmethod
+    def get_dummy_params():
+        """Creates empty parameters for attribute generator initialization."""
+        dataset_params = DatasetParams(0, {})
+        system_params = SystemParams(0, {}, {})
+        dataset_to_system_params = DatasetToSystemParams({}, {}, {}, {})
+        collection_params = CollectionParams({}, {}, 0, {}, 0, {}, 0)
+        data_integrity_params = DataIntegrityParams({})
+        processing_params = ProcessingParams({}, {})
+        connection_params = ConnectionParams(0, 0)
+        return collection_params, dataset_params, system_params, data_integrity_params, processing_params, connection_params
+
+    def test__generate_time(self):
```

Comment:
I'm not a big fan of these double underscores in method names

Reply:
This is a standard in Python unit tests to name test methods "test_" + name of method, and as these methods are private, they start with "_", like "_generate_time", so we get the double underscore in the end.